src/data/nat/basic.lean

```diff
@@ -40,6 +40,19 @@ by rw [← sub_one, nat.sub_sub, one_add]; refl
 
 lemma pred_eq_sub_one (n : ℕ) : pred n = n - 1 := rfl
 
+lemma sub_sub_sub_cancel_right {a b c : ℕ} (h₂ : c ≤ b) : (a - c) - (b - c) = a - b :=
+by rw [nat.sub_sub, ←nat.add_sub_assoc h₂, nat.add_sub_cancel_left]
+
```

Comment:
I would put this near the other lemmas about `sub`

Reply:
Okay, I've moved it, but I'm not confident; there aren't actually a lot of lemmas about sub here, besides interaction with ordering.